### ISSUE TITLE
near account import issue

### ISSUE THREAD

**Issue by hollejacklin** [aurora-is-near/relayer2-public#81]
descript: I used the near cli document account json example to run testnet.yaml, but it shown error of import near account.

test.json 
{
  "account_id": "example-acct.testnet",
  "public_key": "ed25519:7ns2AZVaG8XZrFrgRw7g8qhgddNTN64Zkz7Eo8JBnV5g",
  "private_key": "ed25519:4Ijd3vNUmdWJ4L922BxcsGN1aDrdpvUHEgqLQAUSLmL7S2qE9tYR9fqL6DqabGGDxCSHkKwdaAGNcHJ2Sfd"
}

testnet.yaml
endpoint:
  chainId: 1313161555
  filterFilePath: config/filter.yaml
  engine:
    nearNetworkID: testnet
    nearNodeURL: https://archival-rpc.testnet.near.org
    **signer: example-acct.testnet
    signerKey: config/test.json**
    asyncSendRawTxs: false
    minGasPrice: 0
    minGasLimit: 21000
    gasForNearTxsCall: 300000000000000
    depositForNearTxsCall: 0
    retryWaitTimeMsForNearTxsCall: 3000
    retryNumberForNearTxsCall: 3

error message:
aurora@aurora:~/relayer2-public/out$ ./relayer start --config config/testnet.yaml
panic: runtime error: slice bounds out of range [32:0]

goroutine 1 [running]:
crypto/ed25519.PrivateKey.Public(...)
        /snap/go/10073/src/crypto/ed25519/ed25519.go:58
github.com/aurora-is-near/near-api-go/keystore.LoadKeyPairFromPath({0xc0002a7ec0, 0x10}, {0xc00014a8e8, 0x14})
        /home/aurora/go/pkg/mod/github.com/aurora-is-near/near-api-go@v0.0.13-0.20221011140300-30882c6356c4/keystore/keystore.go:108 +0x64c
github.com/aurora-is-near/near-api-go.LoadAccount(0xc000229ef0, 0xc0000338a8, {0xc00014a8e8, 0x14})
        /home/aurora/go/pkg/mod/github.com/aurora-is-near/near-api-go@v0.0.13-0.20221011140300-30882c6356c4/account.go:43 +0x85
github.com/aurora-is-near/relayer2-public/endpoint.NewEngineEth(0xc00003e240)
        /home/aurora/relayer2-public/endpoint/engineEth.go:45 +0x2a6
main.main.func1(0x0?, {0x0?, 0x0?, 0x0?})
        /home/aurora/relayer2-public/main.go:99 +0xbcd
github.com/aurora-is-near/relayer2-base/cmd.StartCmd.func2(0xc000141b00?, {0xc00037bc80?, 0x2?, 0x2?})
        /home/aurora/go/pkg/mod/github.com/aurora-is-near/relayer2-base@v1.0.0-rc.1/cmd/start.go:22 +0x22
github.com/spf13/cobra.(*Command).execute(0xc000141b00, {0xc00037bc60, 0x2, 0x2})
        /home/aurora/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0xc000141500)
        /home/aurora/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:1040 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/aurora/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:968
main.main()
        /home/aurora/relayer2-public/main.go:167 +0x87

I used mainnet.yaml with my aurora-relayer-V1 account.json file, it shown other error message.

error message:
aurora@aurora:~/relayer2-public/out$ ./relayer start --config config/mainnet.yaml
panic: failed to load Near account

goroutine 1 [running]:
github.com/rs/zerolog.(*Logger).Panic.func1({0x110bca9?, 0x0?})
        /home/aurora/go/pkg/mod/github.com/rs/zerolog@v1.28.0/log.go:375 +0x2d
github.com/rs/zerolog.(*Event).msg(0xc000114300, {0x110bca9, 0x1b})
        /home/aurora/go/pkg/mod/github.com/rs/zerolog@v1.28.0/event.go:156 +0x2d8
github.com/rs/zerolog.(*Event).Msg(0xc00042ff10?, {0x110bca9?, 0xc00042ffa0?})
        /home/aurora/go/pkg/mod/github.com/rs/zerolog@v1.28.0/event.go:108 +0x25
github.com/aurora-is-near/relayer2-public/endpoint.NewEngineEth(0xc000342340)
        /home/aurora/relayer2-public/endpoint/engineEth.go:47 +0x2ed
main.main.func1(0x0?, {0x0?, 0x0?, 0x0?})
        /home/aurora/relayer2-public/main.go:99 +0xbcd
github.com/aurora-is-near/relayer2-base/cmd.StartCmd.func2(0xc000380000?, {0xc000339c80?, 0x2?, 0x2?})
        /home/aurora/go/pkg/mod/github.com/aurora-is-near/relayer2-base@v1.0.0-rc.1/cmd/start.go:22 +0x22
github.com/spf13/cobra.(*Command).execute(0xc000380000, {0xc000339c60, 0x2, 0x2})
        /home/aurora/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0xc000005800)
        /home/aurora/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:1040 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/aurora/go/pkg/mod/github.com/spf13/cobra@v1.6.0/command.go:968
main.main()
        /home/aurora/relayer2-public/main.go:167 +0x87
log message:
{"level":"info","time":"2023-04-02T20:12:22+08:00","message":"opening database with path [/tmp/relayer/data]"}
{"level":"info","time":"2023-04-02T20:12:22+08:00","message":"All 0 tables opened in 0s\n"}
{"level":"info","time":"2023-04-02T20:12:22+08:00","message":"Discard stats nextEmptySlot: 0\n"}
{"level":"info","time":"2023-04-02T20:12:22+08:00","message":"Set nextTxnTs to 0"}
{"level":"info","time":"2023-04-02T20:12:22+08:00","message":"Deleting empty file: /tmp/relayer/data/000039.vlog"}
{"level":"panic","error":**"keystore: public_key does not match private_key: relayer.json"**,"time":"2023-04-02T20:12:22+08:00","message":"failed to load Near account"}
{"level":"info","time":"2023-04-02T20:12:22+08:00","message":"closing database"}
{"level":"info","time":"2023-04-02T20:12:22+08:00","message":"Lifetime L0 stalled for: 0s\n"}
{"level":"info","time":"2023-04-02T20:12:22+08:00","message":"\nLevel 0 [ ]: NumTables: 00. Size: 0 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 64 MiB\nLevel 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB\nLevel 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB\nLevel 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB\nLevel 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB\nLevel 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB\nLevel 6 [B]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB\nLevel Done\n"}

go version
go version go1.20.2 linux/amd64

build info
branch: main
commit: 0cc1f39238c7400d59e0e4bf61a976633863c0a3
tag: v2.0.0-rc.2

**Comment by spilin** [comment]
@hollejacklin Thank you for detailed description, we are looking into this.

**Comment by hollejacklin** [comment]
@tolga-coplu,
I try the nearAPI.KeyPair.fromString('private key here...').publicKey.toString(). it return different public key, and I run the relay normally.

However, I got another error when I call web3 api by javascript to my rpc. 

the error:
Access to XMLHttpRequest at 'http://rpcIP:8545/' from origin 'null' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

I try to modify config, but it was useless.
rpcNode:
  geth:
    HTTPHost: 0.0.0.0
    HTTPPort: 8545
    HTTPVirtualHosts: "*"
    _HTTPCors: "*"_
    HTTPModules:
      - "net"
      - "web3"
      - "eth"
      - "parity"

**Comment by tolga-coplu** [comment]
Hi @hollejacklin,

It seems Relayer2 doesn't add CORS headers to the response of your HTTP OPTIONS request. We are investigating the issue and will inform you when we get progress. 

It would be helpful if you can provide the http request and response including the headers.

**Comment by tolga-coplu** [comment]
@hollejacklin ,
PR #83 created and merged to fix the CORS response headers issue. 
Please file a new issue if you still face any problems.
